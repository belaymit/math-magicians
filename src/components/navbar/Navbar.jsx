import React from 'react';
import { Link } from 'react-router-dom';
import { TbMathFunction } from 'react-icons/tb';
import menuList from './menuList';
import './navbar.scss';

const Navbar = () => (
  <nav>
    <div className="nav-container">
      <a href="/" className="nav-logo">
        <h2>Math Magician</h2>
        <span><TbMathFunction className="icon" /></span>
      </a>
      <ul className="nav-menu">
        {menuList.map((item) => (
          <li key={item.id}>
            <Link to={item.link}>{item.title}</Link>
          </li>
        ))}
      </ul>
    </div>
  </nav>
);

export default Navbar;
