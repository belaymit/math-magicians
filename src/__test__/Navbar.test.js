import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/navbar/Navbar';

require('@testing-library/jest-dom');

describe('Test for Navbar component', () => {
  it('renders Navbar component without any error', () => {
    const output = render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(output).toMatchSnapshot();
  });

  it("renders navbar component with the correct text 'Math Magician' ", () => {
    render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(screen.getByText('Math Magician')).toBeInTheDocument();
  });
});
