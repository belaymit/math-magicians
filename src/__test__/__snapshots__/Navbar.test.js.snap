// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Navbar component renders Navbar component without any error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <div
          class="nav-container"
        >
          <a
            class="nav-logo"
            href="/"
          >
            <h2>
              Math Magician
            </h2>
            <span>
              <svg
                class="icon"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <desc />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M14 10h1c1 0 1 1 2.016 3.527c.984 2.473 .984 3.473 1.984 3.473h1"
                />
                <path
                  d="M13 17c1.5 0 3 -2 4 -3.5s2.5 -3.5 4 -3.5"
                />
                <path
                  d="M3 19c0 1.5 .5 2 2 2s2 -4 3 -9s1.5 -9 3 -9s2 .5 2 2"
                />
                <line
                  x1="5"
                  x2="11"
                  y1="12"
                  y2="12"
                />
              </svg>
            </span>
          </a>
          <ul
            class="nav-menu"
          >
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/calculator"
              >
                Calculator
              </a>
            </li>
            <li>
              <a
                href="/quotes"
              >
                Quotes
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <div
        class="nav-container"
      >
        <a
          class="nav-logo"
          href="/"
        >
          <h2>
            Math Magician
          </h2>
          <span>
            <svg
              class="icon"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <desc />
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M14 10h1c1 0 1 1 2.016 3.527c.984 2.473 .984 3.473 1.984 3.473h1"
              />
              <path
                d="M13 17c1.5 0 3 -2 4 -3.5s2.5 -3.5 4 -3.5"
              />
              <path
                d="M3 19c0 1.5 .5 2 2 2s2 -4 3 -9s1.5 -9 3 -9s2 .5 2 2"
              />
              <line
                x1="5"
                x2="11"
                y1="12"
                y2="12"
              />
            </svg>
          </span>
        </a>
        <ul
          class="nav-menu"
        >
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li>
            <a
              href="/quotes"
            >
              Quotes
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
